apps/bfd-model/bfd-model-rif/src/main/resources/db/scripts/BFD-2572-Cleanup-Medicare-Beneficairyid-History/invoke-psql.sh

```diff
@@ -0,0 +1,329 @@
+             A.BENE_ID,
+             A.MBI_NUM,
+             A.MBI_EFCTV_BGN_DT AS EFCTV_BGN_DT,
+             A.MBI_EFCTV_END_DT AS EFCTV_END_DT
+      FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP A,
+           BENEFICIARIES B
+      WHERE B.BENE_ID = A.BENE_ID
+      AND B.MBI_NUM <> A.MBI_NUM
+   ) V1
+-- find medicare_beneficiaryid_history_temp records that
+-- match beneficiaries_history data already present.
+WHERE (C.BENE_ID = V1.BENE_ID
+AND C.MBI_NUM = V1.MBI_NUM));
+EOM
+
+# STEP_FIVE_INSERT_SQL
+# ==============================================================
+# SQL to extract data from the medicare_beneficiaryid_history_temp
+# and beneficiares tables and create records in the beneficiaries_history
+# table.
+#
+# we want any records that are in the medicare_beneficiaryid_history
+# BUT ARE NOT in the beneficiaries_history table. Also, we'll grab
+# a bene's sex, birth date and HICN info from our current beneficiaries
+# table to have a somewhat more complete set of data for those records
+# inserted into the beneficiaries_history table. We'll use now() for
+# last_updated in case we need to back out the INSERTs.
+# ==============================================================
+read -r -d '' STEP_FIVE_INSERT_SQL << EOM
+INSERT INTO BENEFICIARIES_HISTORY (
+  BENE_HISTORY_ID,
+  BENE_ID,
+  MBI_NUM,
+  EFCTV_BGN_DT,
+  EFCTV_END_DT,
+  BENE_CRNT_HIC_NUM,
+  HICN_UNHASHED,
+  BENE_SEX_IDENT_CD,
+  BENE_BIRTH_DT,
+  LAST_UPDATED)
+SELECT
+  NEXTVAL('beneficiaryhistory_beneficiaryhistoryid_seq'),
+  V2.BENE_ID,
+  V2.MBI_NUM,
+  V2.EFCTV_BGN_DT,
+  V2.EFCTV_END_DT,
+  X.BENE_CRNT_HIC_NUM,
+  X.HICN_UNHASHED,
+  X.BENE_SEX_IDENT_CD,
+  X.BENE_BIRTH_DT,
+  NOW()
+FROM
+  (SELECT TBL1.BENE_MBI_ID,
+      TBL1.BENE_ID,
+      TBL1.MBI_NUM,
+      TBL1.MBI_EFCTV_BGN_DT AS EFCTV_BGN_DT,
+      TBL1.MBI_EFCTV_END_DT AS EFCTV_END_DT,
+      TBL1.LAST_UPDATED
+    FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP TBL1
+    WHERE NOT EXISTS
+        (SELECT V1.*
+          FROM
+            (SELECT DISTINCT
+                A.BENE_ID,
+                A.MBI_NUM,
+                A.EFCTV_BGN_DT
+              FROM BENEFICIARIES_HISTORY A
+              WHERE A.BENE_ID in
+                  (SELECT DISTINCT BENE_ID
+                    FROM MEDICARE_BENEFICIARYID_HISTORY_TEMP)
+                AND A.EFCTV_BGN_DT IS NOT NULL ) V1
+          WHERE TBL1.BENE_ID = V1.BENE_ID
+            AND TBL1.MBI_NUM = V1.MBI_NUM
+            AND V1.EFCTV_BGN_DT IS NOT NULL
+            AND TBL1.MBI_EFCTV_BGN_DT = V1.EFCTV_BGN_DT ) ) V2
+INNER JOIN BENEFICIARIES X ON X.BENE_ID = V2.BENE_ID
+WHERE X.MBI_NUM <> V2.MBI_NUM
+ORDER BY 2, 3;
+EOM
+
+# ==============================
+# BEGIN PROCESSING
+# ==============================
+# belt-and-suspenders to check connection to database.
+echo "Testing db connectivity..."
+now=$(psql -h "${PGHOST}" -U "${PGUSER}" -d "${PGDATABASE}" --quiet --tuples-only -c "select NOW();")
+if [[ "$now" == *"202"* ]]; then
```

Comment:
Guessing this is checking the year snippet? Also guessing whatever comes back from this if it fails can't have a 202 in it?

Reply:
correct.